test/test_jit.py

```diff
@@ -254,6 +254,49 @@ def f(x, y):
         self.assertExpectedGraph(trace)
         self.assertExportImport(trace, (x, y))
 
+    def test_peephole(self):
+        a = torch.tensor([0.4], requires_grad=True)
+        b = torch.tensor([0.7], requires_grad=True)
+        c = torch.tensor([0.4, 0.5], requires_grad=True)
+        d = torch.tensor([0], dtype=torch.int32)
+
+        def f(x, y):
+            return x.type_as(y)
+
+        trace, z = torch.jit.get_trace_graph(f, (a, b))
+        self.run_pass('peephole', trace)
+        self.assertExpectedGraph(trace, subname="same_size")
+        trace, z = torch.jit.get_trace_graph(f, (a, c))
+        self.run_pass('peephole', trace)
+        self.assertExpectedGraph(trace, subname="different_size")
```

Comment:
Sizes don't really matter for this optimization, so why do we check that?

Reply:
I want to make sure type_as is removed even if sizes are different.

Reply:
But why would that matter? Just use tensors of completely mismatched sizes and assert it works then. Going by this reasoning we should also start checking strides, devices, and other properties, but that's too much.